transports/onion/src/lib.rs

```diff
@@ -22,37 +23,95 @@ pub enum OnionError {
     AddrErr(#[from] TorAddrError),
     #[error("error in arti")]
     ArtiErr(#[from] arti_client::Error),
-    #[error("onion services are not implented yet, since arti doesn't support it. (awaiting Arti 1.2.0)")]
-    OnionServiceUnimplemented,
 }
 
 #[derive(Clone)]
-pub struct OnionClient {
+pub struct OnionClient<R: Runtime> {
     // client is in an Arc, because wihtout it the Transport::Dial method can't be implemented,
     // due to lifetime issues. With the, eventual, stabilization of static async traits this issue
     // will be resolved.
-    client: Arc<TorClient<PreferredRuntime>>,
+    client: Arc<TorClient<R>>,
+    safe_mode: bool,
 }
 
-pub type OnionBuilder = TorClientBuilder<PreferredRuntime>;
+pub type OnionBuilder<R> = TorClientBuilder<R>;
 
-impl OnionClient {
+impl<R: Runtime> OnionClient<R> {
     #[inline]
-    pub fn builder() -> OnionBuilder {
-        TorClient::builder()
+    pub fn from_builder(builder: OnionBuilder<R>, safe_mode: bool) -> Result<Self, OnionError> {
```

Comment:
Instead of a bool, can we use an enum here? That would be much more self-documenting. I am thinking something like:

```rust
pub enum AddressConversion {
    DnsOnly,
    IpAndDns
}
```

The rustdoc on this enum can then link to Tor's documentation of `DangerouslyIntoTorAddress` etc which will explain the privacy caveats.